src/bun.js/api/bun/socket.zig

```diff
@@ -1972,7 +1972,11 @@ fn NewSocket(comptime ssl: bool) type {
             _: *JSC.CallFrame,
         ) callconv(.C) JSValue {
             JSC.markBinding(@src());
-            this.socket.open(!this.handlers.is_server);
+            if (comptime ssl) {
+                if (!this.detached) {
+                    this.socket.open(!this.handlers.is_server);
```

Comment:
I don't 100% understand the purpose of this function to begin with

Reply:
this basically starts TLS handshake, in a control manner after we deal with JS stuff internal socket attach